usr.bin/ssh/sshd.c

```diff
@@ -962,33 +955,28 @@ server_accept_loop(int *sock_in, int *sock_out, int *newsock, int *config_s,
 			continue;
 
 		for (i = 0; i < options.max_startups; i++) {
-			if (children[i].configfd == -1 ||
-			    config_pollfd[i] == -1 ||
-			    !(pfd[config_pollfd[i]].revents & (POLLOUT|POLLHUP)))
+			if (children[i].pipefd == -1 ||
+			    startup_pollfd[i] == -1 ||
+			    !(pfd[startup_pollfd[i]].revents & POLLOUT))
+				continue;
+			ptr = sshbuf_mutable_ptr(children[i].config);
+			len = sshbuf_len(children[i].config);
+			ret = write(children[i].pipefd, ptr, len);
+			if (ret == -1 && (errno == EINTR || errno == EAGAIN))
 				continue;
-			ptr = sshbuf_mutable_ptr(rexec_state);
-			len = sshbuf_len(rexec_state);
-			if (children[i].sent >= len) {
-  config_send_failed:
+			if (ret <= 0) {
 				if (children[i].early)
 					listening--;
 				srclimit_done(children[i].pipefd);
 				child_close(&(children[i]), 0, 0);
 				continue;
 			}
-			ptr += children[i].sent;
-			len -= children[i].sent;
-			ret = write(children[i].configfd, ptr, len);
-			if (ret == -1 && (errno == EINTR || errno == EAGAIN))
-				continue;
-			if (ret <= 0)
-				goto config_send_failed;
 			if (ret == len) {
 				/* finished sending config */
-				close(children[i].configfd);
-				children[i].configfd = -1;
+				sshbuf_free(children[i].config);
+				children[i].config = NULL;
 			} else {
-				children[i].sent += ret;
+				sshbuf_consume(children[i].config, ret);
```

Comment:
should check the return status from sshbuf_consume() here :)

Reply:
done